cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -1645,6 +1645,19 @@ class CallInstruction extends Instruction {
    * Gets the number of arguments of the call, including the `this` pointer, if any.
    */
   final int getNumberOfArguments() { result = count(this.getAnArgumentOperand()) }
+
+  /**
+   * Holds if the result is a side effect for the argument at the specified index, or `this` if
+   * `index` is `-1`.
+   *
+   * This helper predicate makes it easy to join on both of these columns at once, avoiding
+   * pathological join orders in case the argument index should get joined first.
+   */
+  pragma[noinline]
+  final SideEffectInstruction getAParameterSideEffect(int index) {
```

Comment:
Good questions. I thought our naming conventions basically limited names to `getFoo()` or `getAFoo()`, rather than `getFoos()`.

And I'll have to actually figure out what the new pragmas do before I feel qualified to start using them:)

Reply:
From https://github.com/github/codeql/blob/main/docs/ql-style-guide.md#naming:
> Predicates that can return multiple results should be named `getA...` or `getAn...`.

We generally talk about QL expressions as representing one possible value rather than the set of possible values.